panoply/resources.py

```diff
@@ -12,5 +12,32 @@ class Resource(TypedDict):
     id: str
     title: str
     fields: Optional[List[Field]]
+    available: Optional[bool]
     required: Optional[bool]
     requires: Optional[List[str]]
+
+
+def list_resources(resources: List[Resource]) -> Optional[List[Dict]]:
+    if resources:
+        return [
+            {"name": resource["title"],
+             "value": resource["id"],
+             "disabled": not resource.get("available", True),
```

Comment:
@zoriana2021, @ilia-panoply, why do we need this inconsistency in field names of resource objects? Why don't we use `title`, `id` and `available` or `name`, `value` and `disabled` everywhere for resource objects? Is it too late to prevent it?

Reply:
Yeah, it is not necessary, but why do we need different property names? Why not to have `name`, `value` and `disabled` on both UI and in the sources and not `name`, `value`, `disabled` on UI and `title`, `id`, `available` in the sources?

Reply:
I agree that it adds some complexity but following discussions we had previously regarding naming conventions it's more reasonable to keep the convention of `id` and `title` same as in source type and source.
this function should be just a helper method for the UI
but now when I think about it...
@zoriana2021 
1. how fields other than `name`, `value` and `disabled` will help us, other fields are not useful on the UI
2. why not to create some generic function for converting any list of objects to a UI-compatible list?

Reply:
@ilia-panoply I think this is exactly the function you're talking about. If we're okay with having different property names on UI and in the sources then I can approve this PR.

Reply:
1. we may need this fields to have the validation functionson the sdk level (for example, parent-children validation)
2. what do you mean by `any list of objects`? The name of the UI may differ for different objects, e.g resource and field.

Reply:
1. why does the validation function need all this info? in case of validating list selections, all you need is the values of the selected items.
2. what do you think about something like that 
`from typing import List, Dict
from collections.abc import Callable


def to_ui_compatible_list(items: List[Dict], create_name: Callable[Dict, str], create_value: Callable[Dict, str]) -> List[Dict]:
    for item in items:
        item['name'] = create_name(item)
        item['value'] = create_value(item)

    return items


items = [
    {
        "a": "a",
        "b": "b",
        "c": "c",
        "d": "d"
    }
]


if __name__ == '__main__':
    print(to_ui_compatible_list(items, lambda item: f"{item['a']}_{item['b']}", lambda item: f"{item['c']}"))`

Reply:
of course you can implement something like that with map

Reply:
@ilia-panoply @kirill-panoply 
I can suggest something like that:

```
def convert_to_ui_format(items, create_name, create_value, 
        is_disabled=lambda x: False, requires=lambda x: []):
    if items:
        return [
            {"name": create_name(item),
             "value": create_value(item),
             "disabled": is_disabled(item),
             "requires": requires(item)
            }
            for item in items]
    else:
        return
```

And then fields would have the following structure

```
convert_to_ui_format(
        fields, 
        create_name = lambda field: field['name'] + ' '+ field['type'] if field.get("type") else field["name"],
        create_value = lambda field: field["name"], 
        is_disabled = lambda field: not field["is_available"]
        )
```

And resource would have
```
convert_to_ui_format(
        resources, 
        create_name = lambda resource: resource["title],
        create_value = lambda resource: resource["id"], 
        is_disabled = lambda resource: not resource["available"],
        requires = lambda resource: resource["requires"]
        )
```

For me it looks like we make the things more complicated instead of simplify them. It seems that it would be easier to write function from scratch than to use sdk utils. 


And regarding the names: Maybe, we should make same names for resources and fields? 
`name` or `title`
`value` or `id`
`is_mandatory` or `required`
`is_available` or `available`
As long as I don't think that it is necessary to take terminology from UI components but I don't see the reason to have different terminology in similar concepts.

Reply:
the difference in the field names is one of the reasons to have such util, 
I agree that it's better to use the same terminology everywhere but currently, it will be an overhead to refactor everything. Please create a task to investigate it, and let's think about it after the validation with Alon